tests/nodeos_eos_evm_gasparam_fork_test.py

```diff
@@ -0,0 +1,1072 @@
+
+    #verify that the non producing node is not alive (and populate the producer nodes with current getInfo data to report if
+    #an error occurs)
+    time.sleep(2.0) # give sometime for the nonProdNode to shutdown
+    if nonProdNode.verifyAlive():
+        Utils.errorExit("Expected the non-producing node to have shutdown.")
+
+    Print("Analyzing the producers leading up to the block after killing the non-producing node, expecting divergence at %d" % (blockNum))
+
+    firstDivergence=analyzeBPs(blockProducers0, blockProducers1, expectDivergence=True)
+    # Nodes should not have diverged till the last block
+    if firstDivergence!=blockNum:
+        Utils.errorExit("Expected to diverge at %s, but diverged at %s." % (firstDivergence, blockNum))
+    blockProducers0=[]
+    blockProducers1=[]
+
+    killBlockNum=blockNum
+    lastBlockNum=killBlockNum+(maxActiveProducers - 1)*inRowCountPerProducer+1  # allow 1st testnet group to produce just 1 more block than the 2nd
+
+    # update gas parameter in minor fork (node0), but not node1
+    Utils.Print("Update gas parameter in minor fork: ram price = 6 EOS per MB, gas price = 10Gwei")
+    trans = prodNode.pushMessage(evmAcc.name, "updtgasparam", json.dumps({"ram_price_mb":"6.0000 EOS","minimum_gas_price":10000000000}), '-p {0}'.format(evmAcc.name), silentErrors=False)
+    prodNode.waitForTransBlockIfNeeded(trans[1], True);
+    time.sleep(2)
+
+    Utils.Print("Transfer funds to trigger gas parameter in minor fork")
+    # EVM -> EOS
+    #   0x9E126C57330FA71556628e0aabd6B6B6783d99fA private key: 0xba8c9ff38e4179748925335a9891b969214b37dc3723a1754b8b849d3eea9ac0
+    toAdd = makeReservedEvmAddress(convert_name_to_value(bobAcc.name))
+    evmSendKey = "ba8c9ff38e4179748925335a9891b969214b37dc3723a1754b8b849d3eea9ac0"
+    Print("Transfer EVM->EOS funds 1Gwei from account %s to %s" % (evmAcc.name, bobAcc.name))
+    nonce = 1
+    gasP = getGasPrice()
+    signed_trx = w3.eth.account.sign_transaction(dict(
+        nonce=nonce,
+        gas=300000,       #300k Gas
+        gasPrice=gasP,
+        to=Web3.to_checksum_address(toAdd),
+        value=int(100000000000000), # 0.0001 EOS = 100,000 Gwei
+        data=b'',
+        chainId=evmChainId
+    ), evmSendKey)
+    actData = {"miner":minerAcc.name, "rlptx":Web3.to_hex(signed_trx.rawTransaction)[2:]}
+
+    # push transaction to node0's minor fork (proda, prodb)
+    trans = prodNode.pushMessage(evmAcc.name, "pushtx", json.dumps(actData), '-p {0}'.format(minerAcc.name), silentErrors=False)
+    time.sleep(1.0)
+    prodNode.waitForTransBlockIfNeeded(trans[1], True)
+    row4=prodNode.getTableRow(evmAcc.name, evmAcc.name, "account", 4) 
+    Utils.Print("\taccount row4 in node0: ", row4)
+    assert(row4["eth_address"] == "9e126c57330fa71556628e0aabd6b6b6783d99fa")
+    assert(row4["balance"] == "0000000000000000000000000000000000000000000000024c84ff8c77eda400")
+
+    # push the same transaction to node1's minor fork (prodc)
```

Comment:
why we are also pushing the tx into the other fork?

My initial idea was to:
 - once the cluster is established
 - set `ConsensusParameter` to `A`
 - verify in the eos-evm-rpc that `A` is what we are getting from `eth_getBlock`
 - trigger a fork
 - set `ConsensusParameter` to `B` (in the minority fork)
 - verify in the eos-evm-rpc that `B` is what we are getting from `eth_getBlock`
 - restore fork
 - verify in the eos-evm-rpc that `A` is what we are getting from `eth_getBlock`

Reply:
the transaction is to ensure the gas consumption calculation is using the current gas parameter values. You can see the account's balances are different for the same transaction in different forks (with different gas parameters). Later on after fork was resolved, the account balance changed in leap & eos-evm-node which means the old blocks (with old gas parameter) was reverted and the new blocks (with new gas parameter) was applied.